### PR TITLE
Mjahmadee patch

### PR DESCRIPTION
**Pull request by MJAHMADEE**
Solve TypeError: 'type' object is not subscriptable
https://stackoverflow.com/a/73427982/10682223

### CODE REVIEW

**Comment by MJAHMADEE** [comment]
> Thanks for your pull request! It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> View this [failed invocation](https://github.com/tensorflow/docs/pull/2116/checks?check_run_id=7935868395) of the CLA check for more information.
> 
> For the most up to date status, view the checks section at the bottom of the pull request.

https://stackoverflow.com/a/73427982/10682223

File: tools/tensorflow_docs/api_generator/reference_resolver.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
+from __future__ import annotations
```
Comment (markmcd) [inline tools/tensorflow_docs/api_generator/reference_resolver.py:15]: This should be the only change you need to make to enable this file to be loaded in Py3.7.

File: tools/tensorflow_docs/api_generator/reference_resolver.py [hunk 2]
```diff
@@ -78,15 +79,11 @@ class ReferenceResolver:
       """,
       flags=re.VERBOSE)
 
-  def __init__(
-      self,
-      *,
```
Comment (markmcd) [inline tools/tensorflow_docs/api_generator/reference_resolver.py:83]: Leave the `*` kwonly operator there please

File: tools/tensorflow_docs/api_generator/reference_resolver.py [hunk 3]
```diff
@@ -78,15 +79,11 @@ class ReferenceResolver:
...
-      duplicate_of: dict[str, str],
-      is_fragment: dict[str, bool],
-      py_module_names: Union[list[str], dict[str, str]],
-      link_prefix: Optional[str] = None,
-      physical_path: Optional[dict[str, str]] = None,
```
Comment (markmcd) [inline tools/tensorflow_docs/api_generator/reference_resolver.py:88]: Leave `physical_path` in place; removing it breaks the method as it's still referenced.

**Review by markmcd — CHANGES_REQUESTED** [review]
By importing the future annotations module, you should be able to leave the new-style type annotations in place.

**Comment by 8bitmp3** [comment]
Closing for now until there's an update following @markmcd 's review and comments (Aug 22)

**Pull request closed without merging**